backend/src/xfd_django/xfd_api/tasks/utils/query_redshift.py

```diff
@@ -46,9 +50,11 @@ def _get_pool():
     return _POOL
 
 
+@cloudwatch_metric()
 def query_redshift(query, params=None):
     """Execute a query on Redshift and return results as list of dicts."""
     pool = _get_pool()
+    query_name = str(query)[:120].replace("\n", " ")
```

Comment:
If query string varies a lot (especially if it includes IDs or user‐supplied data) you risk high dimension cardinality for metric dimension “QueryName”. That may lead to explosion of metric series and cost/performance issues. Which i think in this case my be acceptable since we are try to universally use this function. Just to bring light to the scenario.

Reply:
Ya this query_name is strictly used for logging Titles in the Cloudwatch  Metrics, see emit_redshift_metric() in backend/src/xfd_django/xfd_api/tasks/utils/cloudwatch_metrics.py.